echoshader/dev_class/echo_stats.py

```diff
@@ -0,0 +1,460 @@
+
+        self.hist_cols = 1
+
+        self._sync_widge_stats()
+
+    def _sync_widge_stats(self):
+        """
+        Constructs all the necessary hist widgets attributes
+
+        Parameters
+        ----------
+        None
+
+        Returns
+        -------
+        None
+        """
+        # https://panel.holoviz.org/reference/widgets/Button.html#widgets-gallery-button
+        self.hist_button = panel.widgets.Button(
+            name="Update Hist and Desc Table📊", button_type="primary"
+        )
+
+        # https://panel.holoviz.org/reference/widgets/IntInput.html
+        self.bin_size_input = panel.widgets.IntInput(
+            name="Bin Size Input", value=24, step=10, start=0
+        )
+
+        self.widgets = panel.WidgetBox(
+            self.widgets, panel.WidgetBox(self.bin_size_input, self.hist_button)
+        )
+
+    @param.depends("hist_button.value")
+    def view_table(self):
+        """
+        Create a table describing stats info about box Sv with specific frequency
+
+        Parameters
+        ----------
+        Self
+
+        Returns
+        -------
+        holoviews.Table
+            Show basic stats info, plus skew and kurtosis
+        """
+        time_range = slice(self.box.bounds[0], self.box.bounds[2])
+
+        echo_range = slice(self.box.bounds[3], self.box.bounds[1])
+
+        channel = self.channel_select.value
+
+        # Apply current ranges
+        obj_df = self.MVBS_ds.sel(
+            channel=channel, ping_time=time_range, echo_range=echo_range
+        ).Sv.to_dataframe()
+
+        skew = obj_df["Sv"].skew()
+        kurt = obj_df["Sv"].kurt()
+
+        obj_desc = obj_df.describe().reset_index()
+
+        obj_desc.loc[len(obj_desc)] = ["skew", skew]
+        obj_desc.loc[len(obj_desc)] = ["kurtosis", kurt]
+        obj_desc.loc[len(obj_desc)] = ["Channel", channel]
+
+        # Compute histogram
+        return holoviews.Table(obj_desc.values, "stat", "value").opts(self.table_opts)
+
+    @param.depends("hist_button.value")
+    def view_hist(self):
+        """
+        Create a hist for box Sv with specific frequency
+
+        Parameters
+        ----------
+        Self
+
+        Returns
+        -------
+        hvplot.hist
+        """
+
+        time_range = slice(self.box.bounds[0], self.box.bounds[2])
+
+        echo_range = slice(self.box.bounds[3], self.box.bounds[1])
+
+        channel = self.channel_select.value
+
+        # Apply current ranges
+        obj_df = self.MVBS_ds.sel(
+            channel=channel, ping_time=time_range, echo_range=echo_range
+        ).Sv.to_dataframe()
+        # Compute histogram
+        return obj_df.hvplot.hist("Sv", bins=self.bin_size_input.value).opts(
+            self.hist_opts
+        )
+
+    @param.depends("hist_button.value")
+    def view_all_table(self):
+        """
+        Create a table describing stats info about box Sv with all frequencies
+
+        Parameters
+        ----------
+        Self
```

Comment:
```suggestion
```
`self` wouldn't count as a parameter in this case since this method is part of the class.